producer/writer/consumer_writer.go

```diff
@@ -121,32 +181,63 @@ func (w *consumerWriterImpl) Init() {
 	}()
 }
 
+func (w *consumerWriterImpl) resetConnectionUntilClose() {
+	for {
+		select {
+		case <-w.resetCh:
+			w.resetWithConnectFn(w.connectWithRetry)
+		case <-w.doneCh:
+			w.conn.Close()
+			return
+		}
+	}
+}
+
+func (w *consumerWriterImpl) resetWithConnectFn(fn connectFn) error {
+	// Avoid resetting too frequent.
+	if w.nowFn().UnixNano() < w.lastResetNanos+int64(w.connOpts.ResetDelay()) {
+		return nil
+	}
+	w.m.resetConn.Inc(1)
+	conn, err := fn(w.addr)
+	if err != nil {
+		w.m.resetError.Inc(1)
+		w.logger.Errorf("could not connect to %s, %v", w.addr, err)
+		return err
+	}
+	w.reset(conn)
+	w.logger.Infof("reconnected to %s", w.addr)
```

Comment:
not really, as none of the producers are connecting to too many consumers, if this is logging too much it actually shows some problems

Reply:
I was actually thinking in the case of collectors producing to agg tier and we deploy 50 agg instances at a time and the servers usually take a while to start up so this might be noisy during agg tier deployment. Your call.

Reply:
yeah I think it's probably fine as we don't deploy that often, can remove if it turns out to be too noisy later